lib/fishjam_web/peer_socket.ex

```diff
@@ -22,17 +25,20 @@ defmodule FishjamWeb.PeerSocket do
   end
 
   @impl true
-  def init(state) do
-    {:ok, Map.put(state, :authenticated?, false)}
+  def init(_state) do
+    {:ok, %__MODULE__{}}
   end
 
   @impl true
   def handle_in({encoded_message, [opcode: :binary]}, %{authenticated?: false} = state) do
     case PeerMessage.decode(encoded_message) do
       %PeerMessage{content: {:auth_request, %AuthRequest{token: token}}} ->
         with {:ok, %{peer_id: peer_id, room_id: room_id}} <- PeerToken.verify(token),
-             {:ok, room_pid} <- RoomService.find_room(room_id),
-             :ok <- Room.set_peer_connected(room_id, peer_id),
+             {:ok, node_name} <- get_node_name(room_id),
+             args = [room_id, peer_id, Node.self(), self()],
+             {:ok, connect_result} <-
```

Comment:
Nope, we can't because from this call, we can receive, e.g.: `{:error, room_not_found},` so in the else branch, we would have to added another branch to handle `{:ok, {:error, reason}},` which we could add but IMO it would look bad.